waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -737,6 +771,14 @@ private void buildZookeeperPartitionAssignmentsValidation(ClusterManager cluster
                                 error);
                     PartitionValidationResults partitionValidationResults = partitionValidationResultsList.get(partitionInfo.partitionId);
                     partitionValidationResults.validationResultsMap.put(validationResult.type, validationResult);
+
```

Comment:
I was thinking instead of failing the zk_server test in here, we should probably fail it in buildServerZKPartitionAssignmentsValidation just after `assignmentMatchMap.forEach((partitionId, match) -> {`. For the partitionId, we just check if it's zk_test has failed in step1, if yes we directly fail it's zk_server_test. That way we don't have to use putIfAbsent, we can blindly put (overwrite) the failure message multiple time.

Reply:
I was thinking to only fail server_partition_consistency if zk_test failed because of "partition handled by multiple server" error. Now I am thinking it shouldn't matter. Will make the change as you suggested.